[package] libcurl/8.16.0: On Windows compilation using with_brotli=True fails

### Description

Issue started with 8.12. 
Building without brotli works as expected. On 8.16.0 Failures happens on line 939 of Curl's CMakeLists.txt: 

CMake Error at CMake/Macros.cmake:76 (find_package):
  find_package called with invalid argument "REQUIRED CONFIG"
Call Stack (most recent call first):
  CMakeLists.txt:939 (curl_dependency_option)

It looks like ConanFile.py, is replacing this line in Curl:
https://github.com/conan-io/conan-center-index/blob/fba8c094ce9a29fd84773b5fa4fab8fb9f70c1e4/recipes/libcurl/all/conanfile.py#L322C9-L322C170 

Not sure how to fix that though.


### Package and Environment Details

* Package Name/Version: **libcurl/8.16.0**
* Operating System+version: **Window 11 24H2**
* Compiler+version: **Visual Studio 17.14.16**
* Conan version: **conan 2.5.0**


### Conan profile

[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.version=194
os=Windows


### Steps to reproduce

conan install D:/Build64/Conan/Static/Conan.txt --build missing -s;arch=x86_64;-s;build_type=Debug;-s;compiler.runtime=static

Conan.txt:
```
[requires]
    openssl/3.6.0
    libssh2/1.11.0
    zlib/1.3.1
    libcurl/8.16.0
    libarchive/3.7.9
    brotli/1.1.0
    xz_utils/5.4.5
    eigen/3.4.0
    pcre2/10.44
[options]
    libcurl/*:with_brotli=True
    libcurl/*:with_libssh2=True
    libcurl/*:with_ssl=openssl
    libcurl/*:with_threaded_resolver=False
    libarchive/*:with_lzma=True
    libarchive/*:with_iconv=False
    cyrus-sasl/*:with_gssapi=True
    openldap/*:with_cyrus_sasl=True
    pcre2/*:build_pcre2_16=False
    pcre2/*:build_pcre2_32=False
    pcre2/*:build_pcre2grep=False
    pcre2/*:with_zlib=False
    pcre2/*:with_bzip2=False
[generators]
    CMakeDeps
```


### Logs

<details><summary>Click to expand log</summary>

```
======== Installing packages ========

brotli/1.1.0: Already installed! (1 of 9)

eigen/3.4.0: Already installed! (2 of 9)

pcre2/10.44: Already installed! (3 of 9)

xz_utils/5.4.5: Already installed! (4 of 9)

zlib/1.3.1: Already installed! (5 of 9)

libarchive/3.7.9: Already installed! (6 of 9)

openssl/3.6.0: Already installed! (7 of 9)

libssh2/1.11.0: Already installed! (8 of 9)


-------- Installing package libcurl/8.16.0 (9 of 9) --------
libcurl/8.16.0: Building from source
libcurl/8.16.0: Package libcurl/8.16.0:00af32fa4944bb15809b41f32b115a12005b1e09

libcurl/8.16.0: Copying sources to build folder

libcurl/8.16.0: Building your package in C:\Users\iwant\.conan2\p\b\libcuca10a225765f9\b

libcurl/8.16.0: Calling generate()
libcurl/8.16.0: Generators folder: C:\Users\iwant\.conan2\p\b\libcuca10a225765f9\b\build\generators

libcurl/8.16.0: CMakeToolchain generated: conan_toolchain.cmake

libcurl/8.16.0: CMakeToolchain generated: C:\Users\iwant\.conan2\p\b\libcuca10a225765f9\b\build\generators\CMakePresets.json

libcurl/8.16.0: CMakeToolchain generated: C:\Users\iwant\.conan2\p\b\libcuca10a225765f9\b\src\CMakeUserPresets.json

libcurl/8.16.0: Generating aggregated env files
libcurl/8.16.0: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']

libcurl/8.16.0: Calling build()

libcurl/8.16.0: Running CMake.configure()

libcurl/8.16.0: RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/iwant/.conan2/p/b/libcuca10a225765f9/p" -DENABLE_CURL_MANUAL="OFF" -DCURL_CA_BUNDLE="auto" -DCURL_CA_PATH="auto" -DCURL_CA_FALLBACK="OFF" -DCMAKE_TRY_COMPILE_CONFIGURATION="Debug" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:/Users/iwant/.conan2/p/b/libcuca10a225765f9/b/src"

-- Using CMake version 4.1.2

-- curl version=[8.16.0]

-- Using Conan toolchain: C:/Users/iwant/.conan2/p/b/libcuca10a225765f9/b/build/generators/conan_toolchain.cmake
-- Conan toolchain: CMAKE_GENERATOR_TOOLSET=v143
-- Conan toolchain: Setting CMAKE_MSVC_RUNTIME_LIBRARY=$<$<CONFIG:Debug>:MultiThreadedDebug>

-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF

-- The C compiler identification is MSVC 19.44.35217.0

-- Detecting C compiler ABI info

-- Detecting C compiler ABI info - done

-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.44.35207/bin/Hostx64/x64/cl.exe - skipped

-- Detecting C compile features

-- Detecting C compile features - done

-- CMake platform flags: WIN32 MSVC-1944

-- Performing Test HAVE_WIN32_WINNT

-- Performing Test HAVE_WIN32_WINNT - Success
-- Found _WIN32_WINNT=0x0a00

-- Picky compiler options: -W4 -Wall -wd4061 -wd4191 -wd4255 -wd4464 -wd4548 -wd4574 -wd4668 -wd4710 -wd4711 -wd4746 -wd4774 -wd4820 -wd5045

-- Could NOT find Perl (missing: PERL_EXECUTABLE) 

CMake Warning at CMakeLists.txt:537 (message):
  Perl not found.  Will not build manuals.



-- Conan: Component target declared 'OpenSSL::Crypto'
-- Conan: Component target declared 'OpenSSL::SSL'

-- Conan: Target declared 'openssl::openssl'

-- Conan: Target declared 'ZLIB::ZLIB'

-- Conan: Including build module from 'C:/Users/iwant/.conan2/p/b/opensd74d20cbf40a7/p/lib/cmake/conan-official-openssl-variables.cmake'

-- Looking for OPENSSL_IS_BORINGSSL

-- Looking for OPENSSL_IS_BORINGSSL - not found

-- Looking for OPENSSL_IS_AWSLC

-- Looking for OPENSSL_IS_AWSLC - not found

-- Looking for LIBRESSL_VERSION_NUMBER

-- Looking for LIBRESSL_VERSION_NUMBER - not found

CMake Error at CMake/Macros.cmake:76 (find_package):
  find_package called with invalid argument "REQUIRED CONFIG"
Call Stack (most recent call first):
  CMakeLists.txt:939 (curl_dependency_option)



-- Conan: Component target declared 'Libssh2::libssh2'

-- Check size of size_t

-- Check size of size_t - done

-- Check size of time_t

-- Check size of time_t - done

-- Check size of curl_socket_t

-- Check size of curl_socket_t - done

-- Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss

-- Features: alt-svc HSTS HTTPS-proxy IPv6 Largefile libz NTLM SSL threadsafe TLS-SRP Unicode UnixSockets

-- Enabled SSL backends: OpenSSL v3+

-- Configuring incomplete, errors occurred!



libcurl/8.16.0: ERROR: 
Package '00af32fa4944bb15809b41f32b115a12005b1e09' build failed

libcurl/8.16.0: WARN: Build folder C:\Users\iwant\.conan2\p\b\libcuca10a225765f9\b\build

ERROR: libcurl/8.16.0: Error in build() method, line 233
	cmake.configure()
	ConanException: Error 1 while executing

CMake Error at CMakeLists.txt:325 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  CMakeLists.txt:337 (conan_install)


Configuring incomplete, errors occurred!

```

</details>

--- a/recipes/libcurl/all/conanfile.py
+++ b/recipes/libcurl/all/conanfile.py
@@ -12,7 +12,8 @@ from conan.tools.microsoft import is_msvc, unix_path
 import os
 import re
 
-required_conan_version = ">=2.1.0"
+# INFO: Requires CMakeDeps cmake_extra_variables feature (Conan 2.21)
+required_conan_version = ">=2.21.0"
 
 
 class LibcurlConan(ConanFile):

